numbat/modules/math/functions.nbt

```diff
@@ -37,6 +37,16 @@ fn atanh(x: Scalar) -> Scalar
 
 fn gamma(x: Scalar) -> Scalar
 
+fn _qe_solution<A, B>(a: A, b: B, c: B²/A, sign: Scalar) -> B/A =
+  (-b + sign × sqrt(b² - 4 a c)) / 2 a
+
+fn solve_quadratic_equation<A, B>(a: A, b: B, c: B²/A) -> String =
+  if b² - 4 a c < 0
```

Comment:
It looks like this line gives an incompatible dimensions error. Maybe we can fix it by casting `0`?

Reply:
I tried `0B²` and `0 as B²`, but both gave me `UnknownIdentifier` TypeCheckErrors.

Reply:
You could sidestep the issue and use `b² < 4 a c` instead.

Same thing below for `b² - 4 a c == 0`, can be replaced with `b² == 4 a c`

Reply:
Yes, this is a limitation of the language at the moment, see #37